File: energiai-api/src/main/java/com/hackathon/energiai_api/DTOs/AnalisisResponse.java
package com.hackathon.energiai_api.DTOs;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

public record AnalisisResponse(
        String categoria,
        BigDecimal probabilidad,
        List<String> recomendaciones,
        BigDecimal costo_estimado_mensual,
        Map<String, Integer> clasificacion_equipos,
        String nivel_analisis,
        List<String> campos_imputados,
        List<RecomendacionModelo> recomendaciones_detalle,
        String origen_prediccion,
        String modelo_version,
        List<String> advertencias,
        String nombre_o_numero_analisis
) {
}

File: energiai-frontend/src/app/core/models/analisis-response.ts
export interface RecomendacionDetalle {
  codigo: string;
  texto: string;
  confianza: number | null;
  factores_clave: string[];
}

export interface AnalisisResponse {
  categoria: string;
  probabilidad: number;
  recomendaciones: string[];
  costo_estimado_mensual: number;
  clasificacion_equipos?: Record<string, number>;
  nivel_analisis: 'basico' | 'parcial' | 'avanzado' | 'fallback';
  campos_imputados: string[];
  recomendaciones_detalle?: RecomendacionDetalle[];
  origen_prediccion?: 'modelo_ml' | 'fallback_reglas' | 'registro_legacy';
  modelo_version?: string;
  advertencias?: string[];
  nombre_o_numero_analisis?: string;
}
